src/platform/nrfconnect/FactoryDataParser.c

```diff
@@ -198,70 +234,88 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
         {
             res = res && zcbor_bstr_decode(states, (struct zcbor_string *) &factoryData->vendor_name);
+            updateSize(factoryData, "vendor_name", factoryData->vendor_name.len);
         }
         else if (strncmp("product_name", (const char *) currentString.value, currentString.len) == 0)
         {
             res = res && zcbor_bstr_decode(states, (struct zcbor_string *) &factoryData->product_name);
+            updateSize(factoryData, "product_name", factoryData->product_name.len);
         }
         else if (strncmp("part_number", (const char *) currentString.value, currentString.len) == 0)
         {
             res = res && zcbor_bstr_decode(states, (struct zcbor_string *) &factoryData->part_number);
+            updateSize(factoryData, "part_number", factoryData->part_number.len);
         }
         else if (strncmp("product_url", (const char *) currentString.value, currentString.len) == 0)
         {
             res = res && zcbor_bstr_decode(states, (struct zcbor_string *) &factoryData->product_url);
+            updateSize(factoryData, "product_url", factoryData->product_url.len);
         }
         else if (strncmp("product_label", (const char *) currentString.value, currentString.len) == 0)
         {
             res = res && zcbor_bstr_decode(states, (struct zcbor_string *) &factoryData->product_label);
+            updateSize(factoryData, "product_label", factoryData->product_label.len);
         }
         else if (strncmp("enable_key", (const char *) currentString.value, currentString.len) == 0)
         {
             res = res && zcbor_bstr_decode(states, (struct zcbor_string *) &factoryData->enable_key);
+            updateSize(factoryData, "enable_key", factoryData->enable_key.len);
         }
         else if (strncmp("product_finish", (const char *) currentString.value, currentString.len) == 0)
         {
             res                               = res && uint8_decode(states, &factoryData->product_finish);
             factoryData->productFinishPresent = res;
+            updateSize(factoryData, "product_finish", sizeof(uint32_t));
         }
         else if (strncmp("primary_color", (const char *) currentString.value, currentString.len) == 0)
         {
             res                              = res && uint8_decode(states, &factoryData->primary_color);
             factoryData->primaryColorPresent = res;
+            updateSize(factoryData, "primary_color", sizeof(uint32_t));
         }
         else if (strncmp("user", (const char *) currentString.value, currentString.len) == 0)
         {
             factoryData->user.data = (void *) states->payload;
             res                    = res && zcbor_any_skip(states, NULL);
             factoryData->user.len  = (size_t) ((void *) states->payload - factoryData->user.data);
+            updateSize(factoryData, "user", factoryData->user.len);
```

Comment:
I guess you could use `states->payload - buffer` after `zcbor_list_map_end_force_decode` to learn the total CBOR length, right? Then all these `updateSize` calls wouldn't be needed.

Reply:
Oh, right :)